### PR TITLE
chore: add github ci

### PR DESCRIPTION
**Pull request by saranyailla**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/maven.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # required by wagoid/commitlint-github-action to validate range of commits instead of just the latest one
+      - uses: wagoid/commitlint-github-action@v6
```
Comment (MikeDombo) [inline .github/workflows/maven.yaml:20]: don't need commitlint here, that's fine.

File: .github/workflows/maven.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          distribution: corretto
+          java-version: 8
+          cache: maven
+      - run: rm -rf /tmp/*
+        if: matrix.os == 'ubuntu-latest'
+        continue-on-error: true
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          version: v0.4.0
```
Comment (MikeDombo) [inline .github/workflows/maven.yaml:34]: what's this for?

Comment (saranyailla) [reply .github/workflows/maven.yaml:34]: Without this, latest version of docker buildx is used and that doesn't support `.` in the docker bake file used by the parsec client library. We could cut a PR to fix that but I didn't want to wait on that. So, I've updated the CI to use the version of docker buildx that supports the format. 

https://github.com/parallaxsecond/parsec-client-java/blob/1e44b25e63fdf2aa430c33bdc796afd5c61d5da7/parsec-testcontainers/docker-bake.hcl#L28

```ERROR: docker-bake.hcl:28,8-22: Invalid name; only "[a-zA-Z0-9_-]+" are allowed``` as seen here https://github.com/awslabs/aws-greengrass-labs-parsec-provider/actions/runs/8710795502/job/23893455737#step:7:99

Comment (MikeDombo) [reply .github/workflows/maven.yaml:34]: ok

File: .github/workflows/maven.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Build with Maven
+        run: |
+          git submodule update --init --recursive
```
Comment (MikeDombo) [inline .github/workflows/maven.yaml:37]: you can configure the checkout action to pull submodules instead of doing this

**Comment by MikeDombo** [comment]
Can you enable the actual tests?

**Comment by saranyailla** [comment]
> Can you enable the actual tests?

The tests that are failing are coming from the parsec client sub-module packages. The tests we own are fine.

**Review by MikeDombo — APPROVED** [review]

**Pull request merged**